src/main/resources/super/io/vavr/concurrent/FutureImpl.java

```diff
@@ -148,22 +137,17 @@ private FutureImpl(ExecutorService executorService, Option<Try<T>> value, Queue<
     @Override
     public Future<T> cancel(boolean mayInterruptIfRunning) {
         if (!isCompleted()) {
-            synchronized (lock) {
-                Try.of(() -> job == null || job.cancel(mayInterruptIfRunning))
-                        .recover(ignored -> job != null && job.isCancelled())
-                        .onSuccess(cancelled -> {
-                            if (cancelled) {
-                                this.cancelled = tryComplete(Try.failure(new CancellationException()));
-                            }
-                        });
-            }
+            this.cancelled = tryComplete(Try.failure(new CancellationException()));
         }
         return this;
     }
 
+    private void updateThread() {
+    }
```

Comment:
a private side-effect that does nothing :)

![](https://media.giphy.com/media/YqaqeODWhsbK0/giphy.gif)

Reply:
looks like me :smile:

Reply:
yes 😁